minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-runtime/src/main/java/org/apache/nifi/minifi/configuration/RestChangeNotifier.java

```diff
@@ -0,0 +1,181 @@
+        pippo = new Pippo();
+        this.initialize(properties);
+        serverThread = new Thread(new RunServer());
+    }
+
+    @Override
+    public void initialize(Properties properties) {
+        WebServerSettings webServerSettings = pippo.getServer().getSettings();
+
+        String host = properties.getProperty(HOST_KEY);
+        webServerSettings.host(host != null ? host : "localhost");
+
+        String portRaw = properties.getProperty(PORT_KEY);
+        webServerSettings.port(Integer.parseInt(portRaw != null ? portRaw : "8338"));
+
+        webServerSettings.truststoreFile(properties.getProperty(TRUSTSTORE_LOCATION_KEY));
+        webServerSettings.truststorePassword(properties.getProperty(TRUSTSTORE_PASSWORD_KEY));
+        webServerSettings.keystoreFile(properties.getProperty(KEYSTORE_LOCATION_KEY));
+        webServerSettings.keystorePassword(properties.getProperty(KEYSTORE_PASSWORD_KEY));
+        ((TomcatSettings) webServerSettings).keyAlias(properties.getProperty(KEY_ALIAS_KEY));
+
+        String tomcatHome = properties.getProperty(TOMCAT_HOME_KEY);
+        if(tomcatHome != null) {
+            ((TomcatSettings) webServerSettings).baseFolder(tomcatHome);
+        }
+
+        pippo.getApplication().getPippoSettings().overrideSetting("tomcat.baseFolder","./target/tomcat_home/");
+
+
+        pippo.getApplication().GET("/", new RouteHandler() {
+            @Override
+            public void handle(RouteContext routeContext) {
+                routeContext.send(GET_TEXT);
+            }
+        });
+
+        pippo.getApplication().POST("/upload",  new RouteHandler() {
+            @Override
+            public void handle(RouteContext routeContext) {
+
+                configFile = routeContext.getRequest().getBody();
+                notifyListeners();
+                routeContext.send("Configuration received, notifying listeners");
+            }
+        });
+
+        pippo.getApplication().getRoutePreDispatchListeners().add(new RoutePreDispatchListener() {
+            @Override
+            public void onPreDispatch(Request request, Response response) {
+                log.info(">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>");
+                log.info("request method = " + request.getMethod());
+                log.info("request IP = " + request.getClientIp());
+                log.info("request url = " + request.getUrl());
+                log.info("context path = " + request.getContextPath());
+                log.info("request content type = " + request.getContentType());
+                log.info("<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<");
+            }
+        });
+    }
+
+
+    @Override
+    public Set<ConfigurationChangeListener> getChangeListeners() {
+        return configurationChangeListeners;
+    }
+
+    @Override
+    public boolean registerListener(ConfigurationChangeListener listener) {
+        configurationChangeListeners.add(listener);
```

Comment:
likely want to return the value from the add.  in the case of the Set it will indicate if a no entry was added or if it was already stored in the Set.